docs/madr/2-sqlite.md

```diff
@@ -0,0 +1,89 @@
+# Using aos-sqlite Module
+
+- Status: proposed
+- Deciders: [Dylan], [Ariel], [Phil]
+- Date: 2024-07-22
+- Authors: [Atticus]
+
+## Context and Problem Statement
+
+The `aos-sqlite` module is a WebAssembly (Wasm) binary in the Arweave (ao) ecosystem that extends the functionality of the standard `aos` Wasm binary module. The `aos` module includes an `Eval` handler that consumes Lua strings and uses the `load` method of Lua to execute them. The `aos-sqlite` module is compiled with `lua-sqlite3` from C to Wasm, providing additional SQLite functionality. Given that directly using tables has demonstrated scalability issues, the SQLite implementation offers a potential solution. However, due to its novelty in the ecosystem, there are concerns about its stability in larger products. Therefore, using it here for the ANT Registry provides a suitable environment seperate from other products to test its stability.
+
+## Decision Drivers
+
+The main drivers for this decision are:
+
+- **Scalability**
+  - The current method of using tables has scalability issues that SQLite can potentially address.
+- **Performance**
+  - Improved performance in handling larger datasets with SQLite.
+- **Risk Management**
+  - Mitigating risks in larger products by testing the module in a controlled environment first.
+- **Ecosystem Integration**
+  - Exploring new modules and their potential benefits within the ao ecosystem.
+
+## Considered Options
+
+### Option 1: Continue Using Existing Tables
+
+- **Pros:**
+  - Known and stable.
+  - No need for additional testing or integration.
+- **Cons:**
+  - Demonstrated scalability issues.
+  - Limited performance with larger datasets.
+  - Lack of strictness on data types
+
+### Option 2: Use aos-sqlite Module
+
+- **Pros:**
+  - Addresses scalability issues with direct table usage.
+  - Potential for improved performance with larger datasets.
+  - Maintained by core team of AO
+- **Cons:**
+  - New and relatively untested in the ecosystem.
+  - Potential stability issues in larger products.
```

Comment:
Is there evidence for this?

Reply:
Its on the AO teams org and repo, used by bazar and llama land (fwd research products), and as for reliability issues in larger products/data sets, no, no evidence that I know of. But I don't think anyone has pushed it very far. From my POV the limitations at the moment are mostly going to be ao infra max memory limit on modules. most of them gap it at 1gb I think, despite even wasm32 being able to go to 4gb and wasm64 apparently supporting exabytes.